services/brig/test/integration/API/User/Account.hs

```diff
@@ -1047,6 +1067,30 @@ testDomainsBlockedForRegistration opts brig = withDomainsBlockedForRegistration
   where
     p email = RequestBodyLBS . encode $ SendActivationCode (Left email) Nothing False
 
+-- | FUTUREWORK: This is not very exhaustive.  You can change the flag in
+-- brig.integration.yaml manually and re-run the integration tests for more coverage.  If you
+-- want to have that coverage permanently, grep this module for @setRestrictUserCreation@ and
+-- change all those tests to run with both settings.  Use 'testRestrictedUserCreation' here as
+-- an instructive example.
```

Comment:
I would like to have these tests be more exhaustive. Are we not doing this now because we don't have time?

Reply:
I forgot to mention that I ran the test suite with restricted user creation enabled, and changed all the failing tests [like this](https://github.com/wireapp/wire-server/blob/2c19f78df660a4b2bc783dbc2a729e194d441be6/services/brig/test/integration/API/User/Account.hs#L218).  I went through all tests mentioning "/register" in the description, and verified that just the right tests failed without that change.  So I have tested everything exhaustively, kind of, but I didn't persist that test.

The reason is that it would be a lot of code that isn't only now expensive to write, but will impose a recurring cost on everybody who runs or consults or updates our integration tests.  The code that this would test is literally [these two lines](https://github.com/wireapp/wire-server/blob/2c19f78df660a4b2bc783dbc2a729e194d441be6/services/brig/src/Brig/API/User.hs#L326-L327), and we shouldn't be planning to change those without thinking about how to make this more unit-testable.

I think we shouldn't try to do everything that can be done to improve the quality of our code, but we should think about what's most effective.

Is this a good answer?  Are you ok with leaving the comment now?  If you insist that I resolve this FUTUREWORK, every test case I've extended by the one trivial line will become about 1,5-2 times as long and that much harder to understand and change.